testworld/park.go

```diff
@@ -0,0 +1,248 @@
+// +build testworld
+
+package testworld
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+
+	"testing"
+
+	"time"
+
+	"github.com/centrifuge/go-centrifuge/bootstrap"
+	"github.com/centrifuge/go-centrifuge/cmd"
+	"github.com/centrifuge/go-centrifuge/config"
+	ctx "github.com/centrifuge/go-centrifuge/context"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/node"
+	"github.com/gavv/httpexpect"
+	logging "github.com/ipfs/go-log"
+)
+
+var log = logging.Logger("host")
+
+var hostconfig = []struct {
+	name             string
+	apiPort, p2pPort int64
+}{
+	{"Alice", 8084, 38204},
+	{"Bob", 8085, 38205},
+	{"Charlie", 8086, 38206},
+}
+
+// manager is the manager of the hosts at Testworld (Robert)
+type manager struct {
+
+	// network settings
+	ethNodeUrl, accountKeyPath, accountPassword, network string
+
+	txPoolAccess bool
+
+	// contractAddresses are the addresses of centrifuge contracts on Ethereum
+	contractAddresses *config.SmartContractAddresses
+
+	// bernard is the bootnode for all the hosts
+	bernard *host
+
+	// niceHosts are the happy and nice hosts at the Testworld such as Teddy
+	niceHosts map[string]*host
+
+	// TODO create evil hosts such as William (or Eve)
+
+	// canc is the cancel signal for all hosts
+	canc context.CancelFunc
+}
+
+func newManager(
+	ethNodeUrl, accountKeyPath, accountPassword, network string,
+	txPoolAccess bool,
+	smartContractAddrs *config.SmartContractAddresses) *manager {
+	return &manager{
+		ethNodeUrl:        ethNodeUrl,
+		accountKeyPath:    accountKeyPath,
+		accountPassword:   accountPassword,
+		network:           network,
+		txPoolAccess:      txPoolAccess,
+		contractAddresses: smartContractAddrs,
+		niceHosts:         make(map[string]*host),
+	}
+}
+
+func (r *manager) init() error {
+	cancCtx, canc := context.WithCancel(context.Background())
+	r.bernard = r.createHost("Bernard", 8081, 38201, nil)
+	err := r.bernard.init()
+	if err != nil {
+		return err
+	}
+
+	// start and wait for Bernard since other hosts depend on him
+	go r.bernard.start(cancCtx)
+	time.Sleep(10 * time.Second)
```

Comment:
I am not so deep into it but is there a better way to avoid a `time.sleep` using a channel inside the go routine for example?

Reply:
there is but its much more complicated that.. requires a major refactoring.